app/src/org/commcare/gis/EntityMapUtils.kt

```diff
@@ -25,24 +28,152 @@ object EntityMapUtils {
     }
 
     @Nullable
-    private fun getLatLngFromAddress(address: String): GeoPointData? {
+    private fun getLatLngFromAddress(address: String): LatLng? {
         try {
             if (!address.contentEquals("")) {
                 val data = GeoPointData().cast(UncastData(address))
-                return  data
+                return LatLng(data.latitude, data.longitude)
             }
-        } catch (ignored: IllegalArgumentException) { }
+        } catch (ignored: IllegalArgumentException) {
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun getEntityBoundary(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): Array<LatLng>? {
+        if ("cc_geo_boundary" == detail.templateForms[fieldIndex]) {
+            val boundaryString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parseBoundaryFromString(boundaryString)
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun getEntityBoundaryColor(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): Int? {
+        if ("cc_geo_boundary_color" == detail.templateForms[fieldIndex]) {
+            val colorString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parseHexColor(colorString)
+        }
+        return null
+    }
+
+    @JvmStatic
+    fun getEntityPoints(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): Array<LatLng>? {
+        if ("cc_geo_points" == detail.templateForms[fieldIndex]) {
+            val boundaryString = entity.getFieldString(fieldIndex).trim { it <= ' ' }
+            return parseBoundaryFromString(boundaryString)
+        }
         return null
     }
 
+    @JvmStatic
+    fun getEntityPointColors(
+        entity: Entity<TreeReference>,
+        detail: Detail,
+        fieldIndex: Int
+    ): Array<Int>? {
+        if ("cc_geo_points_colors" == detail.templateForms[fieldIndex]) {
```

Comment:
lets remove any `_points` properties for now based on our discussion this evening.

Reply:
this is back in scope and hence we need to preserve these formats.